src/ray/object_manager/plasma/store.cc

```diff
@@ -286,10 +287,23 @@ uint8_t *PlasmaStore::AllocateMemory(size_t size, MEMFD_TYPE *fd, int64_t *map_s
     *error = PlasmaError::OK;
   }
 
+  // Trigger object spilling if current usage is above the specified threshold.
+  const int64_t footprint_limit = PlasmaAllocator::GetFootprintLimit();
+  if (footprint_limit != 0 && object_spilling_threshold_ < 1.0) {
```

Comment:
```suggestion
  if (footprint_limit != 0) {
```
(more robust)

Reply:
I wanted to make sure by default (threshold at 1.0) we don't need to go into this percentage check so existing behaviors are not changed. Or do you think this is okay?

Reply:
I think this is fine since we should never enter this behavior when at threshold=1 anyway. When we add feature flags, it's best if we can keep the behavior the same without having to check the feature flag at different places in the code.